pkg/provider/pulumi/aws/gateway.go

```diff
@@ -35,29 +37,68 @@ type ApiGateway struct {
 	Api  *apigatewayv2.Api
 }
 
+type nameArnPair struct {
+	name      string
+	invokeArn string
+}
+
 func newApiGateway(ctx *pulumi.Context, name string, args *ApiGatewayArgs, opts ...pulumi.ResourceOption) (*ApiGateway, error) {
 	res := &ApiGateway{Name: name}
 	err := ctx.RegisterComponentResource("nitric:api:AwsApiGateway", name, res, opts...)
 	if err != nil {
 		return nil, err
 	}
 
-	for k, p := range args.OpenAPISpec.Paths {
-		p.Get = awsOperation(p.Get, args.LambdaFunctions)
-		p.Post = awsOperation(p.Post, args.LambdaFunctions)
-		p.Patch = awsOperation(p.Patch, args.LambdaFunctions)
-		p.Put = awsOperation(p.Put, args.LambdaFunctions)
-		p.Delete = awsOperation(p.Delete, args.LambdaFunctions)
-		args.OpenAPISpec.Paths[k] = p
+	nameArnPairs := make([]interface{}, 0, len(args.LambdaFunctions))
+
+	// collect name arn pairs for output iteration
+	for k, v := range args.LambdaFunctions {
+		nameArnPairs = append(nameArnPairs, pulumi.All(k, v.Function.InvokeArn).ApplyT(func(args []interface{}) nameArnPair {
+			name := args[0].(string)
+			arn := args[1].(string)
+			return nameArnPair{
+				name:      name,
+				invokeArn: arn,
+			}
+		}))
 	}
 
-	b, err := args.OpenAPISpec.MarshalJSON()
+	doc := pulumi.All(nameArnPairs...).ApplyT(func(pairs []interface{}) (string, error) {
+		naps := make(map[string]string)
+
+		for _, p := range pairs {
+			if pair, ok := p.(nameArnPair); ok {
+				naps[pair.name] = pair.invokeArn
+			} else {
+				// XXX: Should not occur
+				return "", fmt.Errorf("invalid data")
```

Comment:
could have a more meaningful error here.. 
```suggestion
				return "", fmt.Errorf("invalid data %T %v", p, p)
```

Reply:
done